fiftyone/operators/executor.py

```diff
@@ -615,25 +619,94 @@ def target_view(self, param_name="view_target"):
         Returns:
             a :class:`fiftyone.core.collections.SampleCollection`
         """
-        target = self.params.get(param_name, None)
-        if target == "SELECTED_SAMPLES":
-            return self.view.select(self.selected)
-        if target == "DATASET":
+        target = self.params.get(param_name)
+        if not target:
+            target = (
+                constants.ViewTarget.BASE_VIEW
+                if self.has_generated_view
+                else constants.ViewTarget.DATASET
+            )
+
+        if target == constants.ViewTarget.CURRENT_VIEW:
+            return self.view
+        if target == constants.ViewTarget.DATASET:
             return self.dataset
-        return self.view
+        if target == constants.ViewTarget.BASE_VIEW:
+            return self.view._base_view
+        if target == constants.ViewTarget.SELECTED_SAMPLES:
+            return self.view.select(self.selected)
+        if target == constants.ViewTarget.SELECTED_LABELS:
+            return self.view.select_labels(self.selected_labels)
+        if target == constants.ViewTarget.DATASET_VIEW:
+            return self.dataset.view()
+
+        return self.dataset
+
+    # Alias for common word reversal
+    view_target = target_view
+
+    def _init_has_generated_view(self):
+        stages = self.request_params.get("view", None)
+        generated_view_stages = {
+            f"{stage_class.__module__}.{stage_class.__name__}"
+            for stage_class in (
```

Comment:
Note to self: this will need to be updated when https://github.com/voxel51/fiftyone/pull/4773 is merged